test/Microsoft.SqlTools.ServiceLayer.UnitTests/DisasterRecovery/BackupTests.cs

```diff
@@ -3,14 +3,62 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 //
 
-using System;
-using System.Collections.Generic;
-using System.Linq;
+using Microsoft.SqlTools.ServiceLayer.DisasterRecovery;
+using Microsoft.SqlTools.ServiceLayer.TaskServices;
+using Moq;
 using System.Threading.Tasks;
+using Xunit;
 
 namespace Microsoft.SqlTools.ServiceLayer.UnitTests.DisasterRecovery
 {
     public class BackupTests
     {
+        private TaskMetadata taskMetaData = new TaskMetadata
+        {
+            ServerName = "server name",
+            DatabaseName = "database name",
+            Name = "Backup Database", 
+            IsCancelable = true
+        };
+
+        [Fact]
+        public async Task VerifyCreateAndRunningBackupTask()
+        {
+            using (SqlTaskManager manager = new SqlTaskManager())
+            {
+                var mockUtility = new Mock<IBackupUtilities>();
+                DisasterRecoveryService service = new DisasterRecoveryService(mockUtility.Object);
+                SqlTask sqlTask = manager.CreateTask(this.taskMetaData, service.BackupTask);
+                Assert.NotNull(sqlTask);
+                Task taskToVerify = sqlTask.RunAsync().ContinueWith(Task =>
+                {
+                    Assert.Equal(sqlTask.TaskStatus, SqlTaskStatus.Succeeded);                 
+                });
+
+                await taskToVerify;
+            }
+        }
+
+        [Fact]
+        public async Task CancelBackupTask()
```

Comment:
When I switched to use cancellation token's waithandle, there was slight delay between calling cancel and getting signal. So what happend in the test is that performBackup, which is empty method, was finished before cancel was called. I fixed the test by creating stub perform operation which sleeps for few seconds while waiting for cancel.